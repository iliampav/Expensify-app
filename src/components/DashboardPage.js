import React from 'react';

const ExpenseDashboardPage = () => (
    <div>
        <p> Dashboard Page Content</p>
    </div>
)

export default ExpenseDashboardPage;