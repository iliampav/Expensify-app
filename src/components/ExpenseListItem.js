import React from 'react';
import { Link } from 'react-router-dom';
 
const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    //lista dos itens
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>    
    </div>
)

export default ExpenseListItem