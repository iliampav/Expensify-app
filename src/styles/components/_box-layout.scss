.box-layout {
    background-image: url("../../images/bg.jpg");
}