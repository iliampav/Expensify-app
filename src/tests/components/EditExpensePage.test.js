import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let editExpense, history, removeExpense, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    history = { push: jest.fn() };
    removeExpense = jest.fn();
    wrapper = shallow(<EditExpensePage editExpense={editExpense} history={history} removeExpense={removeExpense} expense={expenses[1]}/>);
})

test('Should render EditExpensePage', () => {
    expect(wrapper).toMatchSnapshot();
})

test('Should Handle edit Expense page', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1]);
})

test('Should handle Remove expenses', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({
        id: expenses[1].id
    });
})