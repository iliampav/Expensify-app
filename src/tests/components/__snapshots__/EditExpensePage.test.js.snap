// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpensePage 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content_container"
    >
      <h1
        className="page-header__title"
      >
        Edit page
      </h1>
    </div>
  </div>
  <div
    className="content_container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 109500,
          "createdAt": -345600000,
          "description": "Rent",
          "id": "2",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button button--remove"
      onClick={[Function]}
    >
      Remove Expense
    </button>
  </div>
</div>
`;
